### PR TITLE
librados: add rados_list_inconsistent_obj_list to provide pg inconsis…

### PR DESCRIPTION
**Pull request by Rjerk**
…tent info

Signed-off-by: Liu Lan <liulan@umcloud.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`

</details>

### CODE REVIEW

File: src/librados/librados_c.cc [hunk 1]
```diff
@@ -666,6 +668,135 @@ extern "C" int _rados_inconsistent_pg_list(rados_t cluster, int64_t pool_id,
...
+    f.open_object_section("shard");
+    auto& osd_shard = shard_info.first;
+    f.dump_int("osd", osd_shard.osd);
+    f.dump_bool("primary", shard_info.second.primary);
+    f.close_section(); // shard
+  }
+  f.close_section(); // shards
+}
+
+extern "C" int _rados_inconsistent_obj_list(rados_t cluster,
```
Comment (tchaikov) [inline src/librados/librados_c.cc]: - `cluster` is not used.

File: src/librados/librados_c.cc [hunk 2]
```diff
@@ -666,6 +668,135 @@ extern "C" int _rados_inconsistent_pg_list(rados_t cluster, int64_t pool_id,
...
+{
+  tracepoint(librados, rados_inconsistent_obj_list_enter, cluster, pg_str, size);
+  librados::PlacementGroup pg;
+  int ret = pg.parse(pg_str);
+  if (!ret) {
+    return -EINVAL;
+  }
+
+  librados::Rados rados;
+  ret = rados.init_with_context(g_ceph_context);
```
Comment (tchaikov) [inline src/librados/librados_c.cc]: this is very unusual. i don't think we should do this kind of setup/teardown (which is missing by the way) in a function like this.

File: src/librados/librados_c.cc [hunk 3]
```diff
@@ -666,6 +668,135 @@ extern "C" int _rados_inconsistent_pg_list(rados_t cluster, int64_t pool_id,
 }
 LIBRADOS_C_API_BASE_DEFAULT(rados_inconsistent_pg_list);
 
+static void dump_inconsistent(const librados::inconsistent_obj_t& inc,
```
Comment (tchaikov) [inline src/librados/librados_c.cc]: i don't think we should use a JSON encoded `inconsistent_obj_t` as part of the interface. probably we can define a C style struct as `inconsistent_obj_t` 's counterpart in C ?

@dzafman what do you think?

**Comment by Rjerk** [comment]
jenkins test make check

File: src/librados/librados_c.cc (hunk 2 above)
Comment (Rjerk) [reply src/librados/librados_c.cc]: right, `rados_ioctx_t` should be passed instead of `rados_t`

File: src/librados/librados_c.cc (hunk 3 above)
Comment (Rjerk) [reply src/librados/librados_c.cc]: It makes sense to me.

**Comment by tchaikov** [comment]
jenkins test make check

File: src/mon/MonCommands.h [hunk 4]
```diff
@@ -424,7 +424,8 @@ COMMAND("fs set " \
         "|standby_count_wanted|session_timeout|session_autoclose" \
         "|allow_standby_replay|down|joinable|min_compat_client " \
 	"name=val,type=CephString "					\
-	"name=yes_i_really_mean_it,type=CephBool,req=false",			\
+	"name=yes_i_really_mean_it,type=CephBool,req=false "		\
+	"name=yes_i_really_really_mean_it,type=CephBool,req=false",	\
```
Comment (sebastian-philipp) [inline src/mon/MonCommands.h]: From a user's POV, having two `yes_i_really_mean_it` flags doesn't really make sense out of context, as it's not clear when to use the first one and when to use the other one. 

And we really need two `yes_i_really_mean_it` flags?

File: src/pybind/mgr/dashboard/controllers/iscsi.py [hunk 5]
```diff
@@ -723,19 +723,36 @@ def _config_to_target(target_iqn, config):
         }
         return target
 
+    @staticmethod
+    def _is_executing(target_iqn):
```
Comment (sebastian-philipp) [inline src/pybind/mgr/dashboard/controllers/iscsi.py]: is this related to the PR?

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/block/iscsi-target-details/iscsi-target-details.component.ts [hunk 6]
```diff
@@ -78,7 +78,12 @@ export class IscsiTargetDetailsComponent implements OnChanges, OnInit {
     this.metadata = { root: this.selectedItem.target_controls };
     const cssClasses = {
       target: {
-        expanded: _.join([Icons.large, Icons.bullseye], ' ')
+        expanded: _.join(
```
Comment (sebastian-philipp) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/block/iscsi-target-details/iscsi-target-details.component.ts]: related to your PR?

File: src/mon/MonCommands.h (hunk 4 above)
Comment (Rjerk) [reply src/mon/MonCommands.h]: Sorry for the rebase problem, so I close this pr.

Thanks for your review, but the codes you doubt are already merged to master branch.
